src/main/java/org/folio/rest/impl/LoanPoliciesAPI.java

```diff
@@ -110,7 +113,103 @@ public void putLoanPolicyStorageLoanPoliciesByLoanPolicyId(
     Handler<AsyncResult<Response>> asyncResultHandler,
     Context vertxContext) {
 
-    MyPgUtil.putUpsert204(LOAN_POLICY_TABLE, entity, loanPolicyId, okapiHeaders, vertxContext,
-        PutLoanPolicyStorageLoanPoliciesByLoanPolicyIdResponse.class, asyncResultHandler);
+    validate(entity, asyncResultHandler, () ->
+      MyPgUtil.putUpsert204(LOAN_POLICY_TABLE, entity, loanPolicyId, okapiHeaders, vertxContext,
+          PutLoanPolicyStorageLoanPoliciesByLoanPolicyIdResponse.class, asyncResultHandler));
+  }
+
+  private void respond422(String field, String value, String message, Handler<AsyncResult<Response>> asyncResultHandler) {
+    Errors errors = ValidationHelper.createValidationErrorMessage(field, "" + value, message);
+    Response response = Response.status(422).header("Content-Type", "application/json").entity(errors).build();
+    asyncResultHandler.handle(Future.succeededFuture(response));
+  }
+
+  /**
+   * @return true if bool is TRUE, false if bool is FALSE or null
+   */
+  private boolean is(Boolean bool) {
+    return Boolean.TRUE.equals(bool);
+  }
+
+  void validate(LoanPolicy loanPolicy, Handler<AsyncResult<Response>> asyncResultHandler, Runnable runIfValid) {
+    final boolean isFixed = loanPolicy.getLoansPolicy() != null &&
+                            loanPolicy.getLoansPolicy().getProfileId() != null &&
+                            loanPolicy.getLoansPolicy().getProfileId().equalsIgnoreCase("Fixed");
```

Comment:
Please can this use the standard two space line continuation indentation rather than horizontal alignment (which is harder to maintain and uses valuable horizontal space)

If there are concerns that this could make this less readable, maybe it could be extracted to a method which would reduce the indentation disparity and raise the abstraction of the current method.

Reply:
Done